src/LuaTransformer.ts

```diff
@@ -3664,16 +3664,32 @@ export class LuaTransformer {
             flags |= tstl.FunctionExpressionFlags.Inline;
         }
 
-        const [transformedBody] = this.transformFunctionBody(node.parameters, body, spreadIdentifier);
+        const [transformedBody, scope] = this.transformFunctionBody(node.parameters, body, spreadIdentifier);
 
-        return tstl.createFunctionExpression(
+        const functionExpression = tstl.createFunctionExpression(
             tstl.createBlock(transformedBody),
             paramNames,
             dotsLiteral,
             spreadIdentifier,
             flags,
             node
         );
+
+        if (ts.isFunctionExpression(node) && node.name && scope.referencedSymbols) {
+            const symbol = this.checker.getSymbolAtLocation(node.name);
+            if (symbol) {
+                const symbolId = this.symbolIds.get(symbol);
+                if (symbolId && scope.referencedSymbols.has(symbolId)) {
```

Comment:
```suggestion
                if (symbolId !== undefined && scope.referencedSymbols.has(symbolId)) {
```

It can't happen in practice, since `genSymbolIdCounter` always starts from `1`, but I think it's better to keep things safe when checking numbers and strings.

We probably should enable `strict-boolean-expressions` after we'll migrate to eslint.

Reply:
Good catch. I'm definitely in favor of enabling `strict-boolean-expressions`.